### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by TordJoranger**
Started work on the Integrationtests. 
One test testing the worker class and queue initialization. This one depends on an actual queue Instance in Azure portal. 

One test testing the HandleRequest method from message request received to adding access to db, given everything is as it should data wise.

### CODE REVIEW

File: src/QueueReceiver.Core/Services/ServiceLocator.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using QueueReceiver.Core.Interfaces;
+
```
Comment (bratli) [inline src/QueueReceiver.Core/Services/ServiceLocator.cs:3]: remove white space

File: tests/QueueReceiver.Core.UnitTests/Services/AccessServiceTests.cs [hunk 2]
```diff
@@ -51,6 +51,35 @@ public async Task HandleRequest_returns_early_if_plant_doesnt_existAsync()
...
+                });
+
+            //Act
+            await _service.HandleRequest(accessInfo);
+
+            //Assert
+            _personProjectService.Verify(_ => _.GiveProjectAccessToPlant(somePersonId, It.IsAny<string>()), Times.Once);
+        }
+
+
```
Comment (bratli) [inline tests/QueueReceiver.Core.UnitTests/Services/AccessServiceTests.cs]: remove whitespace

File: src/QueueReceiver.Core/Interfaces/IPrivilegeService.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Threading.Tasks;
+
+namespace QueueReceiver.Core.Interfaces
+{
+    public interface IPrivilegeService
+    {
+        Task GivePrivlieges(string plantId, long personId);
```
Comment (Cnordbo) [inline src/QueueReceiver.Core/Interfaces/IPrivilegeService.cs]: Typo: `Privileges`

Comment (janid1967) [inline src/QueueReceiver.Core/Interfaces/IPrivilegeService.cs]: spelling error

File: src/QueueReceiver.Core/Models/PersonProjectHistoryOperation.cs [hunk 4]
```diff
@@ -36,7 +36,7 @@ public class PersonProjectHistoryOperation
 
         public PersonProjectHistory PersonProjectHistory { get; internal set; }
 
-#pragma warning disable CS8618 //ORM.
+        #pragma warning disable CS8618 //ORM.
```
Comment (janid1967) [inline src/QueueReceiver.Core/Models/PersonProjectHistoryOperation.cs]: explain better why ... not just only ORM

**Comment by janid1967** [comment]
General comment: async Methods should have postfix Async. Sample await _accessService.HandleRequestAsync(accessInfo);

File: tests/QueueReceiver.IntegrationTests/WorkerServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    [TestClass]
+    public class WorkerServiceTests
+    {
+        private const int MillisecondsDelay = 1000;
+
+        private IConfiguration Configuration { get;  }
+
+        public WorkerServiceTests()
+        {
+            WebRequest.DefaultWebProxy = new WebProxy("http://www-proxy.statoil.no:80");
```
Comment (Cnordbo) [inline tests/QueueReceiver.IntegrationTests/WorkerServiceTests.cs]: Should this be in Configuration ?

**Comment by janid1967** [comment]
General comment: I noticed a class initiating properties in constructor. Properties has internal set; Should those properties be private? Sample PersonProjectHistoryOperation

**Review by Cnordbo — APPROVED** [review]

File: src/QueueReceiver.Core/Services/PersonProjectService.cs [hunk 6]
```diff
@@ -9,28 +9,20 @@ public class PersonProjectService : IPersonProjectService
     {
```
Comment (janid1967) [inline src/QueueReceiver.Core/Services/PersonProjectService.cs:8]: remove using using QueueReceiver.Core.Constants;

File: src/QueueReceiver.Core/Services/ServiceLocator.cs [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+namespace QueueReceiver.Core.Services
+{
+    public sealed class ServiceLocator : IServiceLocator
+    {
+        private readonly IServiceScopeFactory _factory;
+        private IServiceScope? _scope;
+
+        public ServiceLocator(IServiceScopeFactory factory)
+        {
+            _factory = factory;
```
Comment (janid1967) [inline src/QueueReceiver.Core/Services/ServiceLocator.cs:13]: expr body

File: src/QueueReceiver.Core/Services/ServiceLocator.cs [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        public T GetService<T>()
+        {
+            CreateScope();
+
+            return _scope!.ServiceProvider.GetService<T>();
+        }
+
+        public IServiceScope CreateScope()
+        {
+            return _factory.CreateScope();
```
Comment (janid1967) [inline src/QueueReceiver.Core/Services/ServiceLocator.cs]: expr body

File: tests/QueueReceiver.Core.UnitTests/Services/EntryPointServiceTests.cs [hunk 9]
```diff
@@ -28,7 +26,6 @@ public class EntryPointServiceTests
             var accessService = new Mock<IAccessService>();
             var service = new EntryPointService(queueClient, accessService.Object, logger.Object);
 
-            //var accessService = SetupCreateScope(serviceLocator);
             return (service, queueClient, logger, accessService);
```
Comment (janid1967) [inline tests/QueueReceiver.Core.UnitTests/Services/EntryPointServiceTests.cs]: logger is unused. Refactor?

File: tests/QueueReceiver.IntegrationTests/AccessServiceTests.cs [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+                context.Plants.Add(new Plant { InternalGroupId = PlantOid, PlantId = plantId });
+                context.Projects.Add(new Project { ParentProjectId = null, PlantId = plantId });
+                context.UserGroups.Add(new UserGroup { Name = PersonProjectConstants.DefaultUserGroup });
+                context.RestrictionRoles.Add(new RestrictionRole { PlantId = plantId, RestrictionRoleId = PersonProjectConstants.DefaultRestrictionRole });
+                await context.SaveChangesAsync();
+            }
+
+            using (var context = new QueueReceiverServiceContext(options))
+            {
+                var (service, graphMock) = Factory(context);
```
Comment (janid1967) [inline tests/QueueReceiver.IntegrationTests/AccessServiceTests.cs:68]: graphMock not in use

**Review by janid1967 — APPROVED** [review]
Approved with suggestions

**Comment by TordJoranger** [comment]
> General comment: I noticed a class initiating properties in constructor. Properties has internal set; Should those properties be private? Sample PersonProjectHistoryOperation

EntityFramework needs these to be minimum internal when creating entities.

File: tests/QueueReceiver.IntegrationTests/WorkerServiceTests.cs (hunk 5 above)
Comment (TordJoranger) [reply tests/QueueReceiver.IntegrationTests/WorkerServiceTests.cs]: Probably, we have the same in Worker.Program. I've not 100% figure out how to set it properly yet. There is a TODO :)

**Pull request merged**